src/pynwb/icephys.py

```diff
@@ -319,15 +335,492 @@ def __get_row_ids(self, sweep_number):
+             'default': None},
+            {'name': 'electrode_metadata', 'type': dict,
+             'doc': 'Additional electrode metadata to be stored in the electrodes table', 'default': None},
+            {'name': 'stimulus_metadata', 'type': dict,
+             'doc': 'Additional stimulus metadata to be stored in the stimuli table', 'default': None},
+            {'name': 'response_metadata', 'type': dict,
+             'doc': 'Additional resposnse metadata to be stored in the responses table', 'default': None},
+            returns='Integer index of the row that was added to this table',
+            rtype=int,
+            allow_extra=True)
+    def add_recording(self, **kwargs):
+        """
+        Add a single recording to the IntracellularRecordingsTable table.
+
+        Typically, both stimulus and response are expected. However, in some cases only a stimulus
+        or a resposne may be recodred as part of a recording. In this case, None, may be given
+        for either stimulus or response, but not both. Internally, this results in both stimulus
+        and response pointing to the same timeseries, while the start_index and index_count for
+        the invalid series will both be set to -1.
+        """
+        # Get the input data
+        stimulus_start_index, stimulus_index_count, stimulus = popargs('stimulus_start_index',
+                                                                       'stimulus_index_count',
+                                                                       'stimulus',
+                                                                       kwargs)
+        response_start_index, response_index_count, response = popargs('response_start_index',
+                                                                       'response_index_count',
+                                                                       'response',
+                                                                       kwargs)
+        electrode = popargs('electrode', kwargs)
+        # Confirm that we have at least a valid stimulus or response
+        if stimulus is None and response is None:
+            raise ValueError("stimulus and response cannot both be None.")
+
+        # Compute the start and stop index if necessary
+        if stimulus is not None:
+            stimulus_start_index, stimulus_index_count = self.__compute_index(stimulus_start_index,
+                                                                              stimulus_index_count,
+                                                                              stimulus, 'stimulus')
+        if response is not None:
+            response_start_index, response_index_count = self.__compute_index(response_start_index,
+                                                                              response_index_count,
+                                                                              response, 'response')
+
```

Comment:
With the updates to ```__compute_index``` we can always compute the start_index and index_count and ensure they are set correctly.

```suggestion
        # Compute the start and stop index if necessary
        stimulus_start_index, stimulus_index_count = self.__compute_index(stimulus_start_index,
                                                                          stimulus_index_count,
                                                                          stimulus, 'stimulus')
        response_start_index, response_index_count = self.__compute_index(response_start_index,
                                                                          response_index_count,
                                                                          response, 'response')
```